.azure-pipelines/client.yml

```diff
@@ -37,60 +34,90 @@ jobs:
   - script: |
       twine check $(Build.ArtifactStagingDirectory)/*
     displayName: 'Verify Readme'
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifacts'
+    condition: succeededOrFailed()
+
+- job: 'Analyze'
+
+  dependsOn:
+    - 'Build'
+
+  pool:
+    vmImage: 'ubuntu-16.04'
+
+  steps:
+  - task: UsePythonVersion@0
+    displayName: 'Use Python $(PythonVersion)'
+    inputs:
+      versionSpec: '$(PythonVersion)'
 
-  - task: CopyFiles@2
-    displayName: 'Move Dependencies Report to Build Artifacts'
+  - script: |
+     pip install wheel Jinja2
+    displayName: 'Prep Environment'
+
+  - task: PythonScript@0
+    displayName: 'Analyze dependencies'
+    enabled: true
     inputs:
-      SourceFolder: '$(Build.SourcesDirectory)'
-      Contents: dependencies.html
-      TargetFolder: '$(Build.ArtifactStagingDirectory)'
-    condition: always()
+      scriptPath: 'scripts/analyze_deps.py'
+      arguments: '--verbose --out "$(Build.ArtifactStagingDirectory)/dependencies.html"'
+
+  - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+    # ComponentGovernance is currently unable to run on pull requests of public projects. Running on non-PR
+    # builds should be sufficient.
+    condition: and(succeededOrFailed(), ne(variables['Build.Reason'],'PullRequest'))
+    displayName: 'Component Detection'
 
   - task: PublishBuildArtifacts@1
+    # To Publish artifacts generated by the 'Analyze dependencies' task under 'Analyze' job
     displayName: 'Publish Artifacts'
     condition: succeededOrFailed()
 
 - job: 'Generic_Tests'
+
   dependsOn:
-    - 'Build_Target_Packages'
+    - 'Build'
+
   strategy:
     matrix:
       Linux_Python27:
-        os.name: 'Linux'
-        os.vmImage: 'ubuntu-16.04'
-        python.version: '2.7'
+        OSName: 'Linux'
```

Comment:
@scbedd it doesn't even look like we use the OSName anywhere so we should probably just delete it. @mikeharder I see you do use OsName as part of a task description in the JS pipeline. Given that the strategy name is used in the job name I personally don't feel it is worth plumbing this extra information through. What do others think?

Reply:
Yeah. everywhere in the template we use os.name.

Reply:
I must be missing the usages then as I don't see "os.name" or "OSName" being used anywhere in this pipeline.

Reply:
You are correct. It's been updated.